src/server/kernel/wseat.cpp

```diff
@@ -419,15 +454,27 @@ void WSeatPrivate::on_keyboard_key(wlr_keyboard_key_event *event, WInputDevice *
                 text, false, 1, device->qtDevice());
     e.setTimestamp(event->time_msec);
 
-    if (qt_sendShortcutOverrideEvent(focusWindow ? (QObject*)focusWindow.get() : (QObject*)qGuiApp,
-                                     e.timestamp(), e.key(), e.modifiers(),
-                                     e.text(), e.isAutoRepeat(), e.count())) {
-        return;
-    }
-
     if (focusWindow) {
-        QCoreApplication::sendEvent(focusWindow, &e);
+        handleKeyEvent(e);
+        if (et == QEvent::KeyPress && xkb_keymap_key_repeats(keyboard->handle()->keymap, code)) {
+            if (m_repeatKey) {
+                m_repeatTimer.stop();
+            }
+            m_repeatKey = std::make_unique<QKeyEvent>(et, qtkey, keyModifiers, code, event->keycode, keyboard->getModifiers(),
+                text, false, 1, device->qtDevice());
+            m_repeatKey->setTimestamp(event->time_msec);
+            m_repeatTimer.setInterval(keyboard->handle()->repeat_info.delay);
+            m_repeatTimer.start();
+        } else if (et == QEvent::KeyRelease && m_repeatKey && m_repeatKey->nativeScanCode() == code) {
+            m_repeatTimer.stop();
+            m_repeatKey.reset();
+        }
```

Comment:
`else` ?

Reply:
Else nothing to do with auto-repeat.
e.g.
- et==press && key_repeats==false, can be modifier
- et==release, but not repeating or release a modifier